src/main/java/bio/terra/cli/service/utils/WorkspaceManagerService.java

```diff
@@ -393,20 +396,18 @@ public ResourceList enumerateResources(
    * bucket as a referenced resource in the workspace.
    *
    * @param workspaceId the workspace to add the resource to
-   * @param name name of the resource. this is unique across all resources in the workspace
-   * @param description description of the resource
-   * @param cloningInstructions instructions for how to handle the resource when cloning the
-   *     workspace
-   * @param gcsBucketName GCS bucket name
+   * @param resourceToAdd resource definition to add
    * @return the GCS bucket resource object
    */
   public GcpGcsBucketResource createReferencedGcsBucket(
-      UUID workspaceId,
-      String name,
-      String description,
-      CloningInstructionsEnum cloningInstructions,
-      String gcsBucketName) {
-    ReferencedGcpResourceApi referencedGcpResourceApi = new ReferencedGcpResourceApi(apiClient);
+      UUID workspaceId, ResourceDescription resourceToAdd) {
```

Comment:
re: CLI specific objects, using the WSM "result" object model as a creation parameter definitely implies to me that we do want to have CLI specific objects. Follow up PF-729 ticket sounds good, and I'd argue it's maybe medium term instead of long term, but we can see how often we use it and if WM API breaks us.